extension/commands/submitTransaction.ts

```diff
@@ -41,118 +41,131 @@ export async function submitTransaction(evaluate: boolean, treeItem?: Instantiat
-                transientData[key] = Buffer.from(transientData[key]);
-            });
+        } else if (channelPeerInfo.length === 1) {
+            selectPeers = UserInputUtil.DEFAULT;
+        } else {
+            selectPeers = await UserInputUtil.showQuickPick('Select a peer-targeting policy for this transaction', [UserInputUtil.DEFAULT, UserInputUtil.CUSTOM]) as string;
         }
-    } catch (error) {
-        outputAdapter.log(LogType.ERROR, `Error with transaction transient data: ${error.message}`);
-        return;
-    }
-
-    const connection: IFabricClientConnection = FabricConnectionManager.instance().getConnection();
-    const channelPeerInfo: {name: string, mspID: string}[] = await connection.getChannelPeersInfo(channelName);
-
-    let selectPeers: string;
-
-    if (channelPeerInfo.length === 0) {
-        outputAdapter.log(LogType.ERROR, `No channel peers available to target`);
-        return;
-    } else if (channelPeerInfo.length === 1) {
-        selectPeers = UserInputUtil.DEFAULT;
-    } else {
-        selectPeers = await UserInputUtil.showQuickPick('Select a peer-targeting policy for this transaction', [UserInputUtil.DEFAULT, UserInputUtil.CUSTOM]) as string;
-    }
-
-    let peerTargetNames: string[] = [];
-    let peerTargetMessage: string = '';
 
-    if (!selectPeers) {
-        return;
-    } else if (selectPeers === UserInputUtil.CUSTOM) {
+        if (!selectPeers) {
+            return;
+        } else if (selectPeers === UserInputUtil.CUSTOM) {
 
-        const peerTargets: Array<IBlockchainQuickPickItem<string>> = await UserInputUtil.showChannelPeersQuickPick(channelPeerInfo) as Array<IBlockchainQuickPickItem<string>>;
+            const peerTargets: Array<IBlockchainQuickPickItem<string>> = await UserInputUtil.showChannelPeersQuickPick(channelPeerInfo) as Array<IBlockchainQuickPickItem<string>>;
 
-        if (!peerTargets || peerTargets.length === 0) {
-            return;
-        } else {
-            peerTargetNames = peerTargets.map((peer: IBlockchainQuickPickItem<string>) => {
-                return peer.data;
-            });
+            if (!peerTargets || peerTargets.length === 0) {
+                return;
+            } else {
+                peerTargetNames = peerTargets.map((peer: IBlockchainQuickPickItem<string>) => {
+                    return peer.data;
+                });
 
-            peerTargetMessage = ` to peers ${peerTargetNames}`;
+                peerTargetMessage = ` to peers ${peerTargetNames}`;
+            }
         }
+    } else {
+        channelName = transactionObject.channelName;
+        smartContract = transactionObject.smartContract;
+        transactionName = transactionObject.transactionName;
+        namespace = transactionObject.namespace;
+        args = transactionObject.args;
+        transientData = transactionObject.transientData;
+        peerTargetNames = transactionObject.peerTargetNames;
+        peerTargetMessage = '';
```

Comment:
I think if `peerTargetNames` is defined, the `peerTargetMessage` should be the same as on line 156: ```peerTargetMessage = ` to peers ${peerTargetNames}`;```

Reply:
We should probably do some validation of the `transactionObject` object and error if something isn't set (correctly).